src/tag_iterator_async.rs

```diff
@@ -0,0 +1,163 @@
+use std::iter::repeat;
+use std::mem;
+use ebml_iterable_specification::{EbmlSpecification, EbmlTag, Master, TagDataType};
+use futures::{AsyncRead, AsyncReadExt, Stream};
+use crate::error::{TagIteratorError, ToolError};
+use crate::tag_iterator_util::{EBMLSize, ProcessingTag};
+use crate::tag_iterator_util::EBMLSize::Known;
+use crate::tag_iterator_util::ProcessingTag::{EndTag, NextTag};
+use crate::tools;
+
+pub struct TagIteratorAsync<R: AsyncRead + Unpin, TSpec>
```

Comment:
As you mentioned in the PR, it would be good to add some documentation to this struct and its methods